File: windows/BlueLink.App/Session/TransferRetryEligibility.cs
using BlueLink.Domain;

namespace BlueLink.Session;

internal static class TransferRetryEligibility
{
    public static bool Allows(TransferItem item, string? peerId, bool connected, bool trusted) =>
        connected && trusted && !string.IsNullOrWhiteSpace(peerId) &&
        string.Equals(item.PeerId, peerId, StringComparison.OrdinalIgnoreCase) &&
        item.Outgoing && item.Id != Guid.Empty && item.IsRetryableTerminal;
}

File: android/app/src/main/java/com/bluelink/android/session/TransferRetryEligibility.kt
package com.bluelink.android.session

import com.bluelink.android.domain.TransferItem
import com.bluelink.android.domain.TransferStatus

internal object TransferRetryEligibility {
    fun allows(item: TransferItem, peerId: String?, connected: Boolean, trusted: Boolean): Boolean =
        connected && trusted && !peerId.isNullOrBlank() && item.peerId.equals(peerId, ignoreCase = true) &&
            item.outgoing && item.id != java.util.UUID(0, 0) && item.status in setOf(
                TransferStatus.FAILED, TransferStatus.REJECTED, TransferStatus.CANCELED)
}
